### PR TITLE
Basic DWARF testing

### PR DESCRIPTION
**Pull request by kripken**
Verify we emit dwarf with `-gforce_dwarf`, with the proper
sections, and validate some contents from those sections.

Adapted from `test_source_map`.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -7350,6 +7350,93 @@ def source_map_file_loc(name):
...
+
+    # make sure the right sections exist
+    self.assertIn('.debug_abbrev', sections)
+    self.assertIn('.debug_info', sections)
+    self.assertIn('.debug_line', sections)
+    self.assertIn('.debug_str', sections)
+    self.assertIn('.debug_ranges', sections)
+
+    # verify some content in the sections
+    self.assertIn('"src.cpp"', sections['.debug_info'])
```
Comment (dschuff) [inline tests/test_core.py]: do we want to assert anything about the path?

File: tests/test_core.py [hunk 2]
```diff
@@ -7350,6 +7350,93 @@ def source_map_file_loc(name):
...
+
+    # the line section looks like this:
+    # Address            Line   Column File   ISA Discriminator Flags
+    # ------------------ ------ ------ ------ --- ------------- -------------
+    # 0x000000000000000b      5      0      3   0             0  is_stmt
+    src_to_addr = {}
```
Comment (dschuff) [inline tests/test_core.py]: in the presence of optimizations I believe it's possible to have one line/col map to more than one address. Why not `addr_to_source`?

File: tests/test_core.py (hunk 1 above)
Comment (kripken) [reply tests/test_core.py]: Not sure what you mean by "path" here? Do you mean to check the DIE that `"src.cpp"` string is in, or something like that?

Comment (kripken) [reply tests/test_core.py]: Or do you mean the full path to `src.cpp`? It's encoded as a relative address, that's the whole string.

File: tests/test_core.py (hunk 2 above)
Comment (kripken) [reply tests/test_core.py]: Good point. Let me try to figure out a better test that avoids that issue.

File: tests/test_core.py (hunk 1 above)
Comment (dschuff) [reply tests/test_core.py]: yeah that's what I meant. it makes sense this way.

**Comment by kripken** [comment]
Ok, I think the test should be useful now, PTAL.

The test ensures each interesting line is mapped to one
debug line, and verifies that points to a proper place in
the wasm binary. (We track 3 calls that the optimizer
can't reorder or inline, so we can figure out where that is.)

**Pull request merged**